src/components/withAuthentication.js

```diff
@@ -1,27 +1,28 @@
 import React from 'react'
 
 import { auth } from '../firebase/firebase'
-import AuthUserContext from './AuthUserContext'
+import { connect } from 'react-redux'
 
-const withAuthentication = Component =>
+const withAuthentication = Component => {
   class WithAuthentication extends React.Component {
-    state = { authUser: null }
-
     componentDidMount() {
+      const { onAuthChange } = this.props
+
       auth.onAuthStateChanged(authUser => {
-        this.setState({ authUser: authUser || null })
+        onAuthChange(authUser || null)
       })
     }
 
     render() {
-      const { authUser } = this.state
-
-      return (
-        <AuthUserContext.Provider value={authUser}>
-          <Component />
-        </AuthUserContext.Provider>
-      )
+      return <Component />
     }
   }
 
+  const mapDispatchToProps = dispatch => ({
+    onAuthChange: authUser => dispatch({ type: 'AUTH_USER_SET', authUser }),
```

Comment:
We usually create `actionTypes.js` and `actionCreators.js` files. Former has constant strings with action types for specific store, latter has functions that creates an action like you have here.

Then you could create simplified verison of dispatch props:

```js
import { authChange } from 'store/session/actionCreators';
// ...
const mapDispatchToProps = {
  onAuthChange: authChange,
}
```

You could change naming here if you like, or even combine `actionTypes` file with `actionCreators` (having `actions.js` for example)

Reply:
Do you have an example of actionCreator that can be appplied to mapDispatchToProps in that way ?

Reply:
Found something, thanks.